pkg/network/usm/ebpf_javatls.go

```diff
@@ -36,195 +36,176 @@ const (
-	javaAgentAllowRegex = nil
-	javaAgentBlockRegex = nil
-	if c.JavaAgentAllowRegex != "" {
-		javaAgentAllowRegex, err = regexp.Compile(c.JavaAgentAllowRegex)
-		if err != nil {
-			javaAgentAllowRegex = nil
-			log.Errorf("allow regex can't be compiled %s", err)
-		}
-	}
-	if c.JavaAgentBlockRegex != "" {
-		javaAgentBlockRegex, err = regexp.Compile(c.JavaAgentBlockRegex)
-		if err != nil {
-			javaAgentBlockRegex = nil
-			log.Errorf("block regex can't be compiled %s", err)
-		}
-	}
 
-	mon := monitor.GetProcessMonitor()
-	return &JavaTLSProgram{
-		cfg:            c,
-		processMonitor: mon,
+	res = &javaTLSProgram{
+		cfg:                 c,
+		processMonitor:      monitor.GetProcessMonitor(),
+		tracerArguments:     buildTracerArguments(c),
+		tracerJarPath:       javaUSMAgentJarPath,
+		injectionAllowRegex: buildRegex(c.JavaAgentAllowRegex, "allow"),
+		injectionBlockRegex: buildRegex(c.JavaAgentBlockRegex, "block"),
+		procRoot:            util.GetProcRoot(),
 	}
+
+	return res
 }
 
-func (p *JavaTLSProgram) Name() string {
+func (p *javaTLSProgram) Name() string {
 	return "java-tls"
 }
 
-func (p *JavaTLSProgram) IsBuildModeSupported(buildMode) bool {
+func (p *javaTLSProgram) IsBuildModeSupported(buildMode) bool {
 	return true
 }
 
-func (p *JavaTLSProgram) ConfigureManager(m *nettelemetry.Manager) {
-	p.manager = m
-	p.manager.Maps = append(p.manager.Maps, []*manager.Map{
+func (p *javaTLSProgram) ConfigureManager(m *nettelemetry.Manager) {
+	m.Maps = append(m.Maps, []*manager.Map{
 		{Name: javaTLSConnectionsMap},
 	}...)
 
-	p.manager.Probes = append(m.Probes,
+	m.Probes = append(m.Probes,
 		&manager.Probe{ProbeIdentificationPair: manager.ProbeIdentificationPair{
-			EBPFFuncName: "kprobe__do_vfs_ioctl",
+			EBPFFuncName: doVfsIoctlKprobeName,
 			UID:          probeUID,
 		},
 			KProbeMaxActive: maxActive,
 		},
 	)
-	rand.Seed(int64(os.Getpid()) + time.Now().UnixMicro())
-	authID = rand.Int63()
 }
 
-func (p *JavaTLSProgram) ConfigureOptions(options *manager.Options) {
+func (p *javaTLSProgram) ConfigureOptions(options *manager.Options) {
 	options.MapSpecEditors[javaTLSConnectionsMap] = manager.MapSpecEditor{
 		Type:       ebpf.Hash,
 		MaxEntries: p.cfg.MaxTrackedConnections,
 		EditorFlag: manager.EditMaxEntries,
 	}
 	options.MapSpecEditors[javaDomainsToConnectionsMap] = manager.MapSpecEditor{
 		Type:       ebpf.Hash,
-		MaxEntries: uint32(p.cfg.MaxTrackedConnections),
+		MaxEntries: p.cfg.MaxTrackedConnections,
 		EditorFlag: manager.EditMaxEntries,
 	}
 
 	options.ActivatedProbes = append(options.ActivatedProbes,
 		&manager.ProbeSelector{
 			ProbeIdentificationPair: manager.ProbeIdentificationPair{
-				EBPFFuncName: "kprobe__do_vfs_ioctl",
+				EBPFFuncName: doVfsIoctlKprobeName,
 				UID:          probeUID,
 			},
 		})
 }
 
-func (p *JavaTLSProgram) GetAllUndefinedProbes() []manager.ProbeIdentificationPair {
+func (p *javaTLSProgram) GetAllUndefinedProbes() []manager.ProbeIdentificationPair {
```

Comment:
i think `p` is not used here and can be replace with `_` , does it pass the lint without this change btw?